ofborg/src/tasks/build.rs

```diff
@@ -382,11 +392,59 @@ impl notifyworker::SimpleNotifyWorker for BuildWorker {
             .last();
         info!("----->8-----");
 
-        actions.build_finished(status, can_build, cannot_build_attrs);
+        let cachix_result = match &self.cachix {
+            Some(conf) => cache_build(conf.clone(), status.clone(), actions.log_snippet()),
+            None => {
+                info!("Cachix not configured, skipping");
+                false
+            }
+        };
```

Comment:
Since `status` is only used to check if it was a success, there's no reason to make it an argument (and lets us avoid `clone`ing it) -- just check in the `match` branch itself.

```suggestion
        let cachix_result = match &self.cachix {
            Some(conf) if status == BuildStatus::Success => {
                cache_build(conf, actions.log_snippet())
            }
            _ => {
                info!("Skipping Cachix because it was either not configured, or the build failed");

                false
            }
        };
```

We can also avoid `clone`ing the config by making `cache_build` (or `cache_push`, if I have my way) take a reference to a `CachixConfig`.